doc/source/whatsnew/v1.3.0.rst

```diff
@@ -31,6 +31,37 @@ Notable bug fixes
 
 These are bug fixes that might have notable behavior changes.
 
+``Categorical.unique`` now always maintains same dtype as original
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Previously, when calling :meth:`Categorical.unique`, unused categories in the new array
+would be removed, meaning that the dtype of the new array would be different than the
+original, if some categories are not present in the unique array:
```

Comment:
Additional issue is that for non-ordered categorical, now also the order is preserved. Current behaviour:

```
dtype = pd.CategoricalDtype(['bad', 'neutral', 'good'])

In [61]: pd.Categorical(['good', 'good', 'bad', 'bad', 'neutral'], dtype=dtype).unique()
Out[61]: 
['good', 'bad', 'neutral']
Categories (3, object): ['good', 'bad', 'neutral']
```